### PR TITLE
APMRover2: Add support of MANUAL_CONTROL msg

### PR DESCRIPTION
**Pull request by khancyr**
Reverse doesn't works with qgroundcontrol as they don't send negative thrust
Based on @OXINARF work

### CODE REVIEW

File: APMrover2/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1238,6 +1238,26 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
...
+        }
+
+        mavlink_manual_control_t packet;
+        mavlink_msg_manual_control_decode(msg, &packet);
+
+        bool override_active = false;
+        int16_t roll = (packet.y == INT16_MAX) ? 0 : rover.channel_steer->get_radio_min() + (rover.channel_steer->get_radio_max() - rover.channel_steer->get_radio_min()) * (packet.y + 1000) / 2000.0f;
+        int16_t throttle = (packet.z == INT16_MAX) ? 0 : rover.channel_throttle->get_radio_min() + (rover.channel_throttle->get_radio_max() - rover.channel_throttle->get_radio_min()) * (packet.z + 1000) / 2000.0f;
+        override_active |= hal.rcin->set_override(uint8_t(rover.rcmap.roll() - 1), roll);
+        override_active |= hal.rcin->set_override(uint8_t(rover.rcmap.throttle() - 1), throttle);
```
Comment (OXINARF) [inline APMrover2/GCS_Mavlink.cpp]: `override_active` is unused. I think you only want to run the two lines below if it is true, but I'm very far from knowing Rover.

Comment (khancyr) [inline APMrover2/GCS_Mavlink.cpp]: If we look at how it is done on RC_OVERRIDE, it is not need. And I agree. If we received an RC_OVERRIDE or MANUAL_CONTROL message, if the message is valide, it means that we have a connection with the GCS even if the message is empty. 
And set_override return true only if the override is > 0 and on a valid channel but override = -1 is valid too ...It think there is some possible improvements in this logic

**Comment by khancyr** [comment]
remove unused variable on rover side

**Comment by khancyr** [comment]
rebased and ready !

**Comment by khancyr** [comment]
Ping @gmorph @rmackay9 @OXINARF

**Comment by rmackay9** [comment]
Looking good, sorry for the delay!

**Pull request merged**